mod_ci/controllers.py

```diff
@@ -925,6 +928,66 @@ def progress_reporter(test_id, token):
     return "FAIL"
 
 
+def comment_pr(test_id, crash, pr_nr, platform):
+    """
+    Upload the test report to the github PR as comment
+
+    :param test_id: The identity of Test whose report will be uploaded
+    :type test_id: str
+    :crash: whether test results in crash or not
+    :type: boolean
+    :pr_nr: PR number to which test commit is related and comment will be uploaded
+    :type: str
+    """
+    from run import app, log
+    regression_testid_passed = g.db.query(TestResult.regression_test_id).filter(
+        and_(TestResult.test_id == test_id,
+             and_(TestResultFile.test_id == test_id,
+                  and_(TestResult.regression_test_id == TestResultFile.regression_test_id,
+                       and_(TestResultFile.got.is_(None),
+                            TestResult.expected_rc == TestResult.exit_code))))).subquery()
+    passed = g.db.query(label('category_id', Category.id), label(
+        'success', count(regressionTestLinkTable.c.regression_id))).filter(
+        and_(regressionTestLinkTable.c.regression_id.in_(regression_testid_passed),
+             Category.id == regressionTestLinkTable.c.category_id)).group_by(
+                 regressionTestLinkTable.c.category_id).subquery()
+    tot = g.db.query(label('category', Category.name), label('total', count(regressionTestLinkTable.c.regression_id)),
+                     label('success', passed.c.success)).outerjoin(
+        passed, passed.c.category_id == Category.id).filter(
+        Category.id == regressionTestLinkTable.c.category_id).group_by(
+        regressionTestLinkTable.c.category_id).all()
+
+    regression_testid_failed = RegressionTest.query.filter(RegressionTest.id.notin_(regression_testid_passed)).all()
+    template = app.jinja_env.get_or_select_template('ci/pr_comment.txt')
+    message = template.render(tests=tot, failed_tests=regression_testid_failed, test_id=test_id,
+                              crash=crash, platform=platform)
+    log.debug('Github PR Comment Message Created for Test_id: {test_id}'.format(test_id=test_id))
+    try:
+        gh = GitHub(access_token=g.github['bot_token'])
+        repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+        # Pull requests are just issues with code, so github consider pr comments in issues
+        pull_request = repository.issues(pr_nr)
+        comments = pull_request.comments().get()
+        bot_name = g.github['bot_name']
+        comment_id = None
+        for comment in comments:
+            if(comment['user']['login'] == bot_name and platform in comment['body']):
+                comment_id = comment['id']
```

Comment:
Please also add a break; if we found it there's no need to further iterate over the remainder of the list (unless there's a reason for it?)